lib/data_magic/document_builder.rb

```diff
@@ -99,8 +99,8 @@ def calculate(field_name, row, dictionary)
         a, b = Expression.new(expr, field_name).variables
                .map { |c| row[c.to_sym] }
                .map { |value| value == 'NULL' ? nil : value }
-               .map { |c| (fix_field_type(item['type'], c)) }
-        a || b
+               .map { |c| fix_field_type(item['type'] || item[:type], c) }
```

Comment:
I think this is an okay solution for the moment, but clearly we have a bigger problem if we don't know whether we're getting a symbol or a string here.